### ISSUE TITLE
Add full graph option for canvas boundary calculation in Preview/Export

### ISSUE THREAD

**Issue by Kali-Hac** [gephi/gephi#1906]
## Expected Behavior
Could gephi have an export of "png" picture that **keeps the size and positions in  the dynamic network picture**?

I want to keep all nodes being placed in the same positions previously and with the same size when I export the next "png" picture with adding more or less nodes.

To be honest, it's very crucial and useful for us to use the  "png" pictures above to analyze and predict the dynamic network with CNN methods that extract the features accurately.

## Current Behavior
The position and size of nodes will change as the number of nodes changes when I export a "png" picture.

For an example,  when I export a "png" picture with 3 nodes or one node, the size of nodes in the picture will be very large filled in the picture. And when I add more 2 nodes and several edges,  the exported "png" picture scale will change accordingly the nodes' size and positions in the prior picture will change. 

## Possible Solution
I think the excellent development team of gephi can easily add an option method（API）to solve this problem. In my opinion, **using the  gizmo of the final network(the global result of dynamic network also the largest network with the maximum amount of nodes ) will be helpful.**

**The coordinates of each node can be available to get in the gexf  file, it truly reflect the positions of the nodes. But the exported "png" picture only cut out the smallest part of the canva that covers the nodes.Therefore it should display the whole canva of the final network when exporting the previous networks. At the same time, the size of the nodes should change to the size of the final network.**  

Maybe the size of the nodes will be very small even that we can't see them, but it's reasonable because it can show the global network pictures in some dynamic period.  **It's very important for our research and the analysis of other scientists.**

### 1.The first network
![The first network](http://op72m4y17.bkt.clouddn.com/RE1.png)
### 2.Adding more nodes, the size becomes small and the positions changes（it's not what we want）
![Adding more nodes, the size becomes small and the positions changes](http://op72m4y17.bkt.clouddn.com/RE2.png)
### 3.we want to keep the sizes and positions of nodes in this final network（it means that the previous "png" picture will using this size and position）
![we want to keep the sizes and positions of nodes in this final network](http://op72m4y17.bkt.clouddn.com/RE3.png)

## Context
We want to use the dynamic network "png" picture to do a research, it will also crucial for other analysts to get the global "png" picture with the fixed nodes in each period. It will be a very broad application to use these pictures, such as using CNN&LSTM to predict the network. 

## Your Environment

* Version used: Gephi 0.9.2
* Java version: Latest Version
* Operating System: Windows

**Comment by mbastian** [comment]
Hi, I've made some analysis on this request and unfortunately it would require some deeper changes. The current canvas size is based on the preview items only. To solve your problem we would want to calculate the canvas size based on the complete network but only render the filtered network. 

This could potentially be solved better with a plugin. I'll remove the 0.9.3 milestone due to difficulty for now but it can still be revisited lated.

**Comment by mbastian** [comment]
Hey I've implemented this feature and it will be included in the next release. There is now an option to either use the currently visible graph (default) for canvas boundaries or the full graph. This should apply to dynamic graphs as well.

The option was added as a toggle button on Preview. It applies there but also on export.
<img width="519" alt="Screenshot 2022-10-24 at 22 16 34" src="https://user-images.githubusercontent.com/234813/197621175-8204b920-2c38-474a-a165-7ada2d0dcad3.png">

For toolkit users, it can be activated with `PreviewController.setGlobalCanvasSize()`.
